packages/next/build/analysis/get-page-static-info.ts

```diff
@@ -256,7 +272,7 @@ export async function getPageStaticInfo(params: {
 
   const fileContent = (await tryToReadFile(pageFilePath, !isDev)) || ''
   if (
-    /runtime|getStaticProps|getServerSideProps|matcher|unstable_allowDynamic/.test(
+    /runtime|getStaticProps|getServerSideProps|matcher|unstable_allowDynamic|export (let|var|const) config/.test(
```

Comment:
Will we correctly detect here if TypeScript is being used since it it can have a type annotation e.g. `export const config: PageConfig = {}`?